File: compiler/testdata/ruby/test_strings.rb
def hello(name)
  puts "debug message"
  "Hello, " + name
end

def hello_interp(name, age)
  comparative = if age > 40
                  "older"
                else
                  "younger"
                end
  puts "#{name} is #{comparative} than me, age #{age}"
end

def matches_foo(foolike)
  if /foo/ =~ foolike
    puts "got a match"
  end
end

def matches_interp(foo, bar)
  if /foo#{foo}/ =~ bar
    puts "got a match"
  end
end

def extract_third_octet(ip)
  ip.match(/\d{1,3}\.\d{1,3}\.(?<third>\d{1,3})\.\d{1,3}/)["third"]
end

greeting = hello("me")
hello_interp("Steve", 38)
matches_foo("football")
matches_interp(10, "foofoo")
extract_third_octet("127.0.0.1")
terms = %w{foo bar baz}
interp_terms = %W{foo #{"BAR BAZ QUUX"} bar}
puts `man -P cat #{"date"}`

File: compiler/testdata/go/test_strings.go
package main

import (
	"fmt"
	"regexp"

	"github.com/redneckbeard/thanos/stdlib"
)

var patt1 = regexp.MustCompile(`foo`)
var patt2 = regexp.MustCompile(`\d{1,3}\.\d{1,3}\.(?P<third>\d{1,3})\.\d{1,3}`)

func Hello(name string) string {
	fmt.Println("debug message")
	return "Hello, " + name
}
func Hello_interp(name string, age int) {
	var comparative string
	if age > 40 {
		comparative = "older"
	} else {
		comparative = "younger"
	}
	fmt.Printf("%s is %s than me, age %d\n", name, comparative, age)
}
func Matches_foo(foolike string) {
	if patt1.MatchString(foolike) {
		fmt.Println("got a match")
	}
}
func Matches_interp(foo int, bar string) {
	patt, _ := regexp.Compile(fmt.Sprintf(`foo%d`, foo))
	if patt.MatchString(bar) {
		fmt.Println("got a match")
	}
}
func Extract_third_octet(ip string) string {
	return stdlib.NewMatchData(patt2, ip).GetByName("third")
}
func main() {
	greeting := Hello("me")
	Hello_interp("Steve", 38)
	Matches_foo("football")
	Matches_interp(10, "foofoo")
	Extract_third_octet("127.0.0.1")
}
